secret_test.go

```diff
@@ -0,0 +1,79 @@
+package pipeline
+
+import (
+	"encoding/json"
+	"testing"
+
+	"github.com/buildkite/interpolate"
+	"gopkg.in/yaml.v3"
+)
+
+func TestSecretMarshalJSON(t *testing.T) {
+	t.Parallel()
+
+	envVar := "DATABASE_URL"
+	secret := &Secret{
+		Key:                 "DATABASE_URL",
+		EnvironmentVariable: &envVar,
+	}
+
+	want := `{"key":"DATABASE_URL","environment_variable":"DATABASE_URL","RemainingFields":null}`
```

Comment:
Inline processing would mean `"RemainingFields":...` shouldn't be present in the output.

```suggestion
	want := `{"key":"DATABASE_URL","environment_variable":"DATABASE_URL"}`
```

Reply:
```
    secret_test.go:27: json.Marshal(pipeline.Secret{Key:"DATABASE_URL", EnvironmentVariable:"DATABASE_URL", RemainingFields:map[string]interface {}(nil)}) = "{\"key\":\"DATABASE_URL\",\"environment_variable\":\"DATABASE_URL\",\"RemainingFields\":null}", want "{\"environment_variable\":\"DATABASE_URL\",\"key\":\"DATABASE_URL\"}"
```

The test I've written `TestSecretMarshalJSON` is still including `RemainingFields` when using `inlineFriendlyMarshalJSON`. Do I need to add further tags to prevent `RemainingFields` from being included?

Reply:
Hmm, seems a bit weird. This looks the same as the other uses of `inlineFriendlyMarshalJSON`...

Reply:
Ah, the `MarshalJSON` method is on the type `*Secret`, but you're passing a `Secret`. This must be preventing the stdlib JSON package from noticing the `MarshalJSON` method. 
A little bit of `log.Print`-debugging confirms.

The solution: either change `MarshalJSON` like so:

```diff
 // MarshalJSON marshals the secret to JSON.
-func (s *Secret) MarshalJSON() ([]byte, error) {
+func (s Secret) MarshalJSON() ([]byte, error) {
 	return inlineFriendlyMarshalJSON(s)
 }
```

or change the test to use a pointer to a `Secret`:

```diff
-	secret := Secret{
+ 	secret := &Secret{
```

Elsewhere I guess we must just be sticking to pointers everywhere.